infogami/infobase/_dbstore/save.py

```diff
@@ -399,15 +399,15 @@ def group_index(self, index):
         """Groups the index based on table.
         """
         groups = defaultdict(dict)
-        for (table, thing_id, property_id), values in index.iteritems():
+        for (table, thing_id, property_id), values in index.items():
```

Comment:
Used only once.  Python 2 will build a list in memory which will take up more RAM.  Python 3 will be fast.